specification/network/resource-manager/Microsoft.Network/stable/2021-05-01/scenarios/applicationGateway.yaml

```diff
@@ -1,23 +1,18 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/documentation/api-scenario/references/v1.2/schema.json
 scope: ResourceGroup
 
-variables:
-  userAssignedIdentities:
-    type: string
-
-prepareSteps:
-  - step: Generate_Unique_Name
-    armTemplate: ./peArmTemplate/Generate_Unique_name_template.json
-
```

Comment:
to avoid potential name conflict, you can use `prefix` to generate unique string at runtime
```
resourceName:
  type: string
  prefix: foo
```